p4-16/psa/charter/P4_Arch_Charter.mdk

```diff
@@ -0,0 +1,51 @@
+Title         : P4 Architectures Working Group Charter
+
+[TITLE]
+
+# Working Group Name
+
+P4 Architectures Working Group
+
+# Chairs
+
+- Calin Cascaval <cascaval@barefootnetworks.com>
+- Andy Fingerhut <jafingerhut@cisco.com>
+
+# Description of the Working Group
+
+As the P4 language evolved to support an increasingly diverse set of programmable devices, there is
+a need to allow vendors and developers to express the architecture (configuration) of such devices.
+This has been a major improvement in the P4~16~ version of the language, which marked the break
+from the confines of the Protocol Independent Switch Architecture (PISA) and provide the language
+infrastructure to define new functionality using `packages` and `externs`. At the same time, there
+is a need to continue to support PISA and the ability of defining portable programs that span
+multiple devices, allow for composing functionality developed by different parties, and in general,
+provide a common platform that vendors can agree to support as a target to lower the barrier of
+entry to data-plane programming.
+
+The main goal of the P4 Architecture working group is to provide a specification for the Portable
+Switch Architecture (PSA). In addition we are working with the P4 language working group and the P4
+API working group to ensure that the language supports the need of defining new architectures and
+the appropriate APIs are defined to control them. As part of the PSA definition, we are working on
+defining the blue-print of defining collections of portable externs, such that programmers have
```

Comment:
Suggest removing "defining the blue-print of" to make it a little less wordy, I believe without altering the meaning, but not a big deal either way.